File: jdroid-java-core/src/main/java/com/jdroid/java/date/DateConfiguration.java
package com.jdroid.java.date;

import java.util.Date;

public class DateConfiguration {
	
	private static String defaultDateTimeFormat = DateTimeFormat.YYYYMMDDHHMMSSZ;
	private static Date fakeNow;

	public static Date getFakeNow() {
		return fakeNow;
	}
	
	public static void setFakeNow(Date fakeNow) {
		DateConfiguration.fakeNow = fakeNow;
	}
	
	public static Boolean isFakeNow() {
		return fakeNow != null;
	}

	public static String getDefaultDateTimeFormat() {
		return defaultDateTimeFormat;
	}

	public static void setDefaultDateTimeFormat(String defaultDateTimeFormat) {
		DateConfiguration.defaultDateTimeFormat = defaultDateTimeFormat;
	}
}

File: jdroid-java-core/src/main/java/com/jdroid/java/date/DateConfiguration.kt
package com.jdroid.java.date

import java.util.Date

object DateConfiguration {

    private var defaultDateTimeFormat = DateTimeFormat.YYYYMMDDHHMMSSZ
    private var fakeNow: Date? = null

    fun getFakeNow(): Date? {
        return fakeNow
    }

    fun setFakeNow(fakeNow: Date?) {
        DateConfiguration.fakeNow = fakeNow
    }

    fun isFakeNow(): Boolean {
        return fakeNow != null
    }

    fun getDefaultDateTimeFormat(): String {
        return defaultDateTimeFormat
    }

    fun setDefaultDateTimeFormat(defaultDateTimeFormat: String) {
        DateConfiguration.defaultDateTimeFormat = defaultDateTimeFormat
    }
}
